### PR TITLE
Disable warnings for cyclonedds dependency

### PR DESCRIPTION
**Pull request by mering**
Unforatuntely, there is no generic way to disable such warnings (https://github.com/bazel-contrib/rules_foreign_cc/issues/1332) but instead it needs to be done for every `rules_foreign_cc` target explicitly.
I also created https://github.com/eclipse-cyclonedds/cyclonedds/issues/2146 upstream to fix the warning in the first place such that this workaround can be removed in the future.

### CODE REVIEW

File: repositories/cyclonedds.BUILD.bazel [hunk 1]
```diff
@@ -179,6 +179,10 @@ cmake(
         },
         no_match_error = "Unsupported build configuration",
     ),
+    copts = [
+        "-Wno-documentation-deprecated-sync",
+        "-Wno-error",
```
Comment (mvukov) [inline repositories/cyclonedds.BUILD.bazel:184]: Which compiler complains about this? Shall we just set `-w`?

Comment (mering) [reply repositories/cyclonedds.BUILD.bazel:184]: LLVM 19, I am fine with `-w` as well.

Comment (mvukov) [reply repositories/cyclonedds.BUILD.bazel:184]: OK, let's go with "-w" then.

Comment (mering) [reply repositories/cyclonedds.BUILD.bazel]: Done.

**Review by mvukov — APPROVED** [review]

**Pull request merged**